libs/components/tabs/src/lib/modules/tabs/tab-button.component.ts

```diff
@@ -45,14 +56,65 @@ export class SkyTabButtonComponent {
   public disabled: boolean;
 
   @Input()
-  public tabStyle: SkyTabsetStyle;
+  public tabIndex: SkyTabIndex;
+
+  @Input()
+  public get tabStyle(): SkyTabsetStyle {
+    return this.#_tabStyle;
+  }
+
+  public set tabStyle(style: SkyTabsetStyle) {
+    this.#_tabStyle = style;
+    this.elementRole = style === 'tabs' ? 'tab' : undefined;
+  }
 
   @Output()
   public buttonClick = new EventEmitter<void>();
 
   @Output()
   public closeClick = new EventEmitter<void>();
 
+  constructor(
+    elementRef: ElementRef,
+    adapterService: SkyTabButtonAdapterService,
+    changeDetectorRef: ChangeDetectorRef,
+    tabsetService: SkyTabsetService
+  ) {
+    this.#adapterService = adapterService;
+    this.#changeDetectorRef = changeDetectorRef;
+    this.#elementRef = elementRef;
+    this.#tabsetService = tabsetService;
+  }
+
+  public elementRole: string | undefined = 'tab';
+  public closeBtnTabIndex = '-1';
+
+  #_tabStyle: SkyTabsetStyle;
+  #adapterService: SkyTabButtonAdapterService;
+  #changeDetectorRef: ChangeDetectorRef;
+  #elementRef: ElementRef;
+  #tabsetService: SkyTabsetService;
+  #ngUnsubscribe = new Subject<void>();
+
+  public ngAfterViewInit(): void {
+    this.#tabsetService.focusedTabBtnIndex
+      .pipe(distinctUntilChanged(), takeUntil(this.#ngUnsubscribe))
+      .subscribe((focusedIndex) => {
+        if (focusedIndex === this.tabIndex) {
+          this.closeBtnTabIndex = '0';
+          this.focusBtn();
+        } else {
+          this.closeBtnTabIndex = '-1';
```

Comment:
you can only tab to the close button if that tab is currently focused (does not have to be active)